extensions/notebook/src/test/book/book.test.ts

```diff
@@ -567,6 +569,35 @@ describe('BooksTreeViewTests', function () {
 			should(notebookUri.scheme).equal('untitled', 'Failed to get untitled uri of the resource');
 		});
 
+		it('openMarkdown should open markdown in the editor', async () => {
+			let executeCommandSpy = sinon.spy(vscode.commands, 'executeCommand');
+			let notebookPath = path.join(rootFolderPath, 'content', 'readme.md');
+			bookTreeViewProvider.openMarkdown(notebookPath);
+			should(executeCommandSpy.calledWith('markdown.showPreview')).be.true('openMarkdown should have called markdown.showPreview');
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		});
+
+		it('openNotebook should open notebook in the editor', async () => {
+			let notebookPath = path.join(rootFolderPath, 'content', 'notebook2.ipynb');
+			await bookTreeViewProvider.openNotebook(notebookPath);
+			should(azdata.nb.notebookDocuments.length).equal(1, 'Should have opened the notebook in the editor.');
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		});
+
+		it('openNotebookAsUntitled should open a notebook as untitled file in the editor', async () => {
```

Comment:
These test names don't seem to match up - for example this is openNotebookAsUntitled but it appears to be testing openExternalLink?

Reply:
Thanks Charles for catching this, there was some mix up with copy paste and the tests for jumbled.